framework/src/operators/map_results.rs

```diff
@@ -0,0 +1,135 @@
+use super::act::Act;
+use super::iterator::*;
+use super::packet_batch::PacketBatch;
+use super::Batch;
+use common::*;
+use headers::EndOffset;
+use interface::Packet;
+use interface::PacketTx;
+use std::marker::PhantomData;
+
+pub type MapFnResult<T, M> = Box<FnMut(&Packet<T, M>) -> Result<()> + Send>;
+
+pub struct MapResults<T, V>
+where
+    T: EndOffset,
+    V: Batch + BatchIterator<Header = T> + Act,
+{
+    parent: V,
+    transformer: MapFnResult<T, V::Metadata>,
+    applied: bool,
+    remove: Vec<usize>,
+    phantom_t: PhantomData<T>,
+}
+
+impl<T, V> MapResults<T, V>
+where
+    T: EndOffset,
+    V: Batch + BatchIterator<Header = T> + Act,
+{
+    pub fn new(parent: V, transformer: MapFnResult<T, V::Metadata>) -> MapResults<T, V> {
+        let capacity = parent.capacity() as usize;
+        MapResults {
+            parent: parent,
+            transformer: transformer,
+            applied: false,
+            remove: Vec::with_capacity(capacity),
+            phantom_t: PhantomData,
+        }
+    }
+}
+
+impl<T, V> Batch for MapResults<T, V>
+where
+    T: EndOffset,
+    V: Batch + BatchIterator<Header = T> + Act,
+{
+}
+
+impl<T, V> Act for MapResults<T, V>
+where
+    T: EndOffset,
+    V: Batch + BatchIterator<Header = T> + Act,
+{
+    #[inline]
+    fn act(&mut self) {
+        if !self.applied {
+            self.parent.act();
+            {
+                let iter = PayloadEnumerator::<T, V::Metadata>::new(&mut self.parent);
+                while let Some(ParsedDescriptor { packet, index: idx }) =
+                    iter.next(&mut self.parent)
+                {
+                    if let Err(ref e) = (self.transformer)(&packet) {
+                        err!(e);
+                        self.remove.push(idx)
+                    }
+                }
+            }
+            self.applied = true;
+
+            if !self.remove.is_empty() {
+                self.parent
+                    .drop_packets(&self.remove[..])
```

Comment:
This is great! Love it. I wouldn't mind if we just replace the old `map` and `transform` with this version as the default and only way to do it.

Reply:
for another time, related to backwards-compat, but maybe just do this as part of the rest of the work toward result operations.